### DISCUSSION TITLE
schedule to only process the dashboard (possibly misunderstanding some basics)

### DISCUSSION THREAD

**Discussion by fracai** [renovatebot/renovate#23613] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

renovate/renovate:36.25.3

### If you're self-hosting Renovate, select which platform you are using.

Gitea or Forgejo

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I'm running Renovate with a nightly Woodpecker cron entry. I will also sometimes run the cron pipeline manually to respond to merged PRs and Dashboard actions. Each run checks the configured repositories, updates the PRs and Dashboard, and then exits.

Ideally I'd have Renovate running constantly so Dashboard actions are handled frequently, but only check for package updates nightly.

I think I can accomplish that with a scheduling rule (I noted the documentation stated that the dashboard isn't constrained by the schedule), but I'm not sure what that schedule would look like. If I increase the cron schedule to run every minute, I'd very quickly hit the Docker Hub rate limit. I can add a schedule to restrict all updates to something like "before 4am", but that would still be over 200 update checks that would take more than a minute to complete anyway.

Am I approaching this from the wrong angle? Is there a way to run Renovate as it's own docker process where it's constantly running to process the dashboard and will perform a single set of dependency updates in the configured schedule? I'd still want to force an update check on demand at times.

For example, if there was an option to disable all update checks (while leaving the Dashboard processing intact) I could schedule that one frequently and leave the default with update checks scheduled for nightly. I didn't see anything like that.

Would it be an option to set a schedule like "before 1am" and run that task frequently _except_ between midnight and one and that's when I'd run the regular Renovate action?

I figure there's probably a cleaner way.

### What you tried so far.

I've tried nothing and I'm all out of ideas.

Basically, I've read a lot of documentation, but I must be missing something or approaching this from a misconception about how it's supposed to function.

Thanks for your help.

**Comment by rarkins** [comment, marked as answer]
If you run Renovate every hour with a schedule of "before 1am" then 23 of 24 hours of the day it will update the dashboard but not create any new PRs. If anyone ticks a dashboard checkbox during those hours then the PR will be created the next time Renovate runs on the repo

**Comment by fracai** [comment]
If it's updating the dashboard outside the schedule, are there any docker actions that could hit the rate limit? Actually, are there ever any actions that would hit the rate limit? Does just fetching meta-data for version checks count against the Docker rate limit? I'll have to look in to that.

If Renovate runs for longer than the schedule "allows", will it cancel operations early? For example, if it's configured to run every hour with that "before 1am" schedule, but the midnight run takes longer than an hour, will it cancel actions at 1am? Or is the schedule just a limit on when the task starts up. I suppose overlapping runs could cause problems either way.

**Reply by rarkins** [reply]
You can hit a rate limit.

Avoid overlapping runs - schedule them further apart if a full run takes more than 1hr

**Reply by fracai** [reply]
Alright, with that I think I'm all set. Thanks for your help.
